src/Array.ts

```diff
@@ -734,6 +735,17 @@ export const rotate = <A>(n: number, xs: Array<A>): Array<A> => {
   }
 }
 
+/**
+ * Remove duplicate entries from an array, keeping the first found item.
+ * @function
+ * @since 1.3.0
+ */
+export const uniq = <A>(as: Array<A>, S: Setoid<A>): Array<A> => {
```

Comment:
@alex-ketch thanks for this PR

> I tried to hazard a guess to the @since tag

1.3.0 is ok

> I was going to use toArray from Set, but that function orders the elements which I personally find counter-intuitive

The reason why `Set.toArray` requires an `Ord` is that otherwise you might end up with different outputs given the same inputs

```ts
export const toArray = <A>(x: Set<A>): Array<A> => {
  const r: Array<A> = []
  x.forEach(e => r.push(e))
  // without sorting
  return r
}

const x = new Set([1, 2])
const y = new Set([2, 1])
console.log(toArray(x)) // [1, 2]
console.log(toArray(y)) // [2, 1]
```

`x = y` but `toArray(x) != toArray(y)`

**Signature**

I propose the following signature which is more aligned to the convention used so far

```ts
export const uniq = <A>(S: Setoid<A>) => (as: Array<A>): Array<A> => {
  ...
}
```

**Dependency on Set**

I know there will be some code duplication but could we avoid a dependency on the `Set` module?

Reply:
I'm concerned about the behavior regarding preserving the relative order of unique elements.

Specifically Purescript (which is the example we should follow) does preserve relative order:

https://github.com/purescript/purescript-arrays/blob/e502257ce0a89ef7bbf89b20665e6f7a491b781f/test/Test/Data/Array.purs#L313-L318

Reply:
@alex-ketch @gcanti 
I'm concerned about not preserving relative order of elements.

Also, note that purescript (that is a reference of the project) does that specifically: 
https://github.com/purescript/purescript-arrays/blob/e502257ce0a89ef7bbf89b20665e6f7a491b781f/test/Test/Data/Array.purs#L313-L318

Reply:
@gcanti The currying and removing dependency on `Set.fromArray` makes sense 👍

---

@sledorze You're talking about `toArray`, correct?
If so I would agree, since as far as I'm aware `Set` makes no guarantees about the order.

> x = y but toArray(x) != toArray(y)

The above is not the case, unless I'm missing something.
https://repl.it/repls/WornSubduedOs

This may be a topic for another issue though

Reply:
@alex-ketch I mean `=` and `!=` in the mathematical sense, i.e. `new Set([1, 2])` and `new Set([2, 1])` represents the same set

```
{ 1, 2 }
```

however `toArray` would output two different arrays: `[1, 2]` is not equal to `[2, 1]` based on the standard equivalence relation defined on `Array<number>` (see its [Setoid instance](https://github.com/gcanti/fp-ts/blob/78917aada5d30f177090141202a3dff7fcd0c77b/src/Array.ts#L58))